addons/web_editor/static/src/js/widgets/widgets.js

```diff
@@ -550,6 +565,126 @@ var ImageWidget = MediaWidget.extend({
+
+        if (url.match(regex)) {
+            for (var i in format) {
+                var endsWith = url.endsWith(format[i]);
+                if (endsWith) {
+                    acceptedURL = true;
+                    $warning.toggleClass('d-none', true);
+                    $success.toggleClass('d-none', false);
+                    this._nameButton('image', 'images');
+                }
+            }
+        }
+        if (!acceptedURL) {
+            $warning.toggleClass('d-none', false);
+            $success.toggleClass('d-none', true);
+            this._nameButton('document', 'images');
+        }
+        if (url.length === 0) {
+            $warning.toggleClass('d-none', true);
+            $success.toggleClass('d-none', true);
+            this._nameButton('', 'images');
+        }
+    },
+    /**
+     * @private
+     */
+    _checkDocumentUrl: function () {
+        var url = this.$('#documenturl').val();
+        var $error = this.$('.error-url-document');
+        var $success = this.$('.success-url-document');
+        var $btn = this.$('.o_upload_document_url_button');
+        var regex = /^(http:\/\/www\.|https:\/\/www\.|http:\/\/|https:\/\/)?[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\.[a-z]{2,5}(:[0-9]{1,5})?(\/.*)?$/;
+        var acceptedURL = false;
+
+        if (url.match(regex)) {
+            acceptedURL = true;
+            $success.toggleClass('d-none', false);
+            $error.toggleClass('d-none', true);
+            this._nameButton('document', 'documents');
+            $btn.removeClass('disabled');
+        }
+        if (!acceptedURL) {
+            $success.toggleClass('d-none', true);
+            $error.toggleClass('d-none', false);
+            this._nameButton('document', 'documents');
+            $btn.addClass('disabled');
+        }
+        if (url.length === 0) {
+            $success.toggleClass('d-none', true);
+            $error.toggleClass('d-none', true);
+            this._nameButton('', 'documents');
+            $btn.toggleClass('disabled');
+        }
+    },
+    /**
+     * @private
+     */
+    _nameButton: function (name, tab) {
+        var $imageUrlBtn = this.$('.o_upload_image_url_button');
+        var $documentUrlBtn = this.$('.o_upload_document_url_button');
+        if (tab === 'images') {
+            if (name === 'document') {
+                name = _t("Add as document");
+            } else if (name === 'image') {
+                name = _t("Add as image");
+            } else {
+                name = _t("Add an image URL");
+            }
+            $imageUrlBtn.html(name);
+        } else {
+            if (name === 'document') {
+                name = _t("Add document");
+            } else {
+                name = _t("Add a document URL");
+            }
+            $documentUrlBtn.html(name);
+        }
+    },
+    /**
+     * @private
+     */
+    _hideUnsplash: function () {
```

Comment:
Still unsplash here

Reply:
hideUnsplash in widgets.js as the purpose is to show Image related stuff and vice versa (hideImage in unsplash_image_widget.js)..

Reply:
No... I do not know all your specs and code but there should not be a reference to unsplash here. I suppose you must do something like "showImages" and override it in unsplash to hideUnsplash there.